### PR TITLE
8250804: Can't set the application icon image for Unity WM on Linux.

### PR DESCRIPTION
**Pull request by azvegint**
As of now we can have two icons: on the side dock and on the top bar.
But this test was written for an old time Unity with an icon in the side dock only. 

Icon at the top by default is displayed as grayscale(the actual reason of submitting bug report).
BTW it may not be presented depending on Gnome Shell settings.

`setIconImage()` seems to be working as expected in this case, but we need to clarify the test's instruction.

Besides that the test has another issue: it does not fail on pressing "Fail" button. It is also fixed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8250804](https://bugs.openjdk.java.net/browse/JDK-8250804): Can't set the application icon image for Unity WM on Linux.

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)
 * [Pankaj Bansal](https://openjdk.java.net/census#pbansal) (@pankaj-bansal - **Reviewer**)
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2838/head:pull/2838`
`$ git checkout pull/2838`

### CODE REVIEW

File: test/jdk/ProblemList.txt [hunk 1]
```diff
@@ -852,7 +852,6 @@ java/awt/dnd/DnDFileGroupDescriptor/DnDFileGroupDescriptor.html 8080185 macosx-a
 javax/swing/JTabbedPane/4666224/bug4666224.html 8144124  macosx-all
 java/awt/event/MouseEvent/AltGraphModifierTest/AltGraphModifierTest.java 8162380 generic-all
 java/awt/Window/TranslucentJAppletTest/TranslucentJAppletTest.java 8163086 macosx-all
-java/awt/image/multiresolution/MultiResolutionIcon/IconTest.java 8250804 macosx-all,linux-all
```
Comment (mrserb) [inline test/jdk/ProblemList.txt:855]: Does this test work fine on the macOS? Looks like it was reported here:

https://bugs.openjdk.java.net/browse/JDK-8250804?focusedCommentId=14379402&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14379402

Comment (mrserb) [reply test/jdk/ProblemList.txt:855]: O, I see that you made it Linux only.

**Review by mrserb — APPROVED** [review]

File: test/jdk/ProblemList.txt (hunk 1 above)
Comment (azvegint) [reply test/jdk/ProblemList.txt:855]: Yes, it does not work on macOS.
For changing dock icon on macOS you should use `java.awt.Taskbar#setIconImage` or assign it with application bundle.

**Review by pankaj-bansal — APPROVED** [review]

**Review by azuev-java — APPROVED** [review]

**Pull request closed without merging**